tests/PE/sections/empty-pe32-dll.test

```diff
@@ -0,0 +1,88 @@
+# RUN: %yaml2obj %s -o %t.obj
+# RUN: %bloaty --raw-map %t.obj | %FileCheck %s
+
+--- !COFF
+OptionalHeader:
+  AddressOfEntryPoint: 4096
+  ImageBase:       268435456
+  SectionAlignment: 4096
+  FileAlignment:   512
+  MajorOperatingSystemVersion: 4
+  MinorOperatingSystemVersion: 0
+  MajorImageVersion: 0
+  MinorImageVersion: 0
+  MajorSubsystemVersion: 4
+  MinorSubsystemVersion: 0
+  Subsystem:       IMAGE_SUBSYSTEM_WINDOWS_GUI
+  DLLCharacteristics: [  ]
+  SizeOfStackReserve: 1048576
+  SizeOfStackCommit: 4096
+  SizeOfHeapReserve: 1048576
+  SizeOfHeapCommit: 4096
+  ExportTable:
+    RelativeVirtualAddress: 0
+    Size:            0
+  ImportTable:
+    RelativeVirtualAddress: 0
+    Size:            0
+  ResourceTable:
+    RelativeVirtualAddress: 0
+    Size:            0
+  ExceptionTable:
+    RelativeVirtualAddress: 0
+    Size:            0
+  CertificateTable:
+    RelativeVirtualAddress: 0
+    Size:            0
+  BaseRelocationTable:
+    RelativeVirtualAddress: 0
+    Size:            0
+  Debug:
+    RelativeVirtualAddress: 0
+    Size:            0
+  Architecture:
+    RelativeVirtualAddress: 0
+    Size:            0
+  GlobalPtr:
+    RelativeVirtualAddress: 0
+    Size:            0
+  TlsTable:
+    RelativeVirtualAddress: 0
+    Size:            0
+  LoadConfigTable:
+    RelativeVirtualAddress: 0
+    Size:            0
+  BoundImport:
+    RelativeVirtualAddress: 0
+    Size:            0
+  IAT:
+    RelativeVirtualAddress: 0
+    Size:            0
+  DelayImportDescriptor:
+    RelativeVirtualAddress: 0
+    Size:            0
+  ClrRuntimeHeader:
+    RelativeVirtualAddress: 0
+    Size:            0
+header:
+  Machine:         IMAGE_FILE_MACHINE_I386
+  Characteristics: [ IMAGE_FILE_EXECUTABLE_IMAGE, IMAGE_FILE_LINE_NUMS_STRIPPED, IMAGE_FILE_LOCAL_SYMS_STRIPPED, IMAGE_FILE_32BIT_MACHINE, IMAGE_FILE_DEBUG_STRIPPED, IMAGE_FILE_DLL ]
+sections:
+  - Name:            .text
+    Characteristics: [ IMAGE_SCN_CNT_CODE, IMAGE_SCN_MEM_EXECUTE, IMAGE_SCN_MEM_READ ]
+    VirtualAddress:  4096
+    VirtualSize:     24
+    SectionData:     5589E581EC0000000090B801000000E900000000C9C20C00
+symbols:         []
+...
+
+# CHECK: FILE MAP:
+# CHECK: 000-1a0	        416		[PE Headers]
```

Comment:
For ELF we try to attribute section headers to their corresponding sections. For example, the section header for `.text` is attributed to `.text` instead of `[ELF Headers]`.

Would this be possible for PE?

Reply:
As outlined in #220, it can be split into:
* DOS Header
* DOS Stub
* Rich Header? (not sure if this can be done reliable, otherwise it will be dumped on top of the DOS Stub)
* NT header
* Per-section header (for each .text, .rdata etc section)

Do you just want the Per-section header here, or also the other DOS/NT header split?

Reply:
and do you want that only for `segments`, or also for `sections` ?

Reply:
I think mainly the per-section headers should be attributed to their respective sections. The other headers sound like they are not specific to any particular section, so `[PE Headers]` sounds like a good fallback for them.

For ELF I've recently changed segments to work the same way: each segment header is attributed to its respective segment.

Reply:
Option 1:
```
Maps for empty-pe64-exe.test.tmp.obj:

FILE MAP:
000-188         392             [PE Headers]
188-1b0          40             .text Header
1b0-1d8          40             .pdata Header
1d8-200          40             [Unmapped]
200-400         512             .text
400-600         512             .pdata

VM MAP:
0000-0188               392             [PE Headers]
0188-01b0                40             .text Header
01b0-01d8                40             .pdata Header
01d8-1000              3624             [-- Nothing mapped --]
1000-1020                32             .text
1020-2000              4064             [-- Nothing mapped --]
2000-200c                12             .pdata
```

Or option 2:
```
Maps for empty-pe64-exe.test.tmp.obj:

FILE MAP:
000-040          64             DOS Header
040-080          64             DOS Stub
080-188         264             NT Header
188-1b0          40             .text Header
1b0-1d8          40             .pdata Header
1d8-200          40             [Unmapped]
200-400         512             .text
400-600         512             .pdata

VM MAP:
0000-0040                64             DOS Header
0040-0080                64             DOS Stub
0080-0188               264             NT Header
0188-01b0                40             .text Header
01b0-01d8                40             .pdata Header
01d8-1000              3624             [-- Nothing mapped --]
1000-1020                32             .text
1020-2000              4064             [-- Nothing mapped --]
2000-200c                12             .pdata
```

Reply:
Thanks for trying these out. After playing with the ELF support, I've change my mind on this (sorry for the change). I think I want to change ELF to use:

- [ELF Header] (for the fixed-size ELF header at the top of the file)
- [ELF Section Headers] (used for all section headers)
- [ELF Program Headers] (used for all program headers)

For PE I think this would mean:

- [PE Headers] (for DOS and NT headers)
- [PE Section Headers] (for all section headers)
- [PE Image Headers] (for the "optional" header, only present in images)

This would be similar to your "option 1", except collapse `.text Header` and `.pdata Header` into `[PE Section Headers]`.

Rationale: each individual section header is not big enough to be interesting, but seeing them all grouped together under `[PE Section Headers]` can help users know how much file space in total they are spending for having N sections in their file.